keanu-project/src/main/java/io/improbable/keanu/distributions/continuous/Dirichlet.java

```diff
@@ -0,0 +1,62 @@
+package io.improbable.keanu.distributions.continuous;
+
+import io.improbable.keanu.distributions.ContinuousDistribution;
+import io.improbable.keanu.distributions.dual.Diffs;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+
+import static io.improbable.keanu.distributions.dual.Diffs.C;
+import static io.improbable.keanu.distributions.dual.Diffs.X;
+
+public class Dirichlet implements ContinuousDistribution {
+
+    private static final double EPSILON =  0.00001;
+    private final DoubleTensor concentration;
+
+    public static ContinuousDistribution withParameters(DoubleTensor concentration) {
+        return new Dirichlet(concentration);
+    }
+
+    private Dirichlet(DoubleTensor concentration) {
+        this.concentration = concentration;
+    }
+
+    @Override
+    public DoubleTensor sample(int[] shape, KeanuRandom random) {
+       final ContinuousDistribution gamma = Gamma.withParameters(
+           DoubleTensor.zeros(shape),
+           DoubleTensor.ones(shape),
+           concentration
+        );
+        final DoubleTensor gammaSamples = gamma.sample(concentration.getShape(), random);
+        return normalise(gammaSamples);
+    }
+
+    @Override
+    public DoubleTensor logProb(DoubleTensor x) {
+        if (Math.abs(x.sum() - 1.0) > EPSILON) {
+            throw new IllegalArgumentException("Sum of dirichlet values to calculate likelihood for must equal 1.");
+        }
+        final double sumConcentrationLogged = concentration.minus(1.).timesInPlace(x.log()).sum();
```

Comment:
This might be a stupid question, but can `x` have 0's here?

Reply:
Yep x can be equal to zero but can't go negative. At zero it should go to negative infinity